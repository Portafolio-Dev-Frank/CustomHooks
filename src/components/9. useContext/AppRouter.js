import React from 'react'
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Link,
    Redirect
} from "react-router-dom";
import { AboutPage } from './AboutPage';
import { HomePage } from './HomePage';
import { LoginPage } from './LoginPage';
import { NavBar } from './NavBar';

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar></NavBar>
                <div className="container">
                    <Switch>
                        <Route exact path="/about" component={AboutPage}></Route>
                        <Route exact path="/login" component={LoginPage}></Route>
                        <Route exact path="/" component={HomePage}></Route>
                        {/* <Route component={HomePage}></Route> */}
                        <Redirect to="/"></Redirect>
                    </Switch>
                </div>
            </div>
        </Router>
    )
}
