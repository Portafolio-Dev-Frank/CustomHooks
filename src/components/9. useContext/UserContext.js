import { createContext } from 'react';

// HIGH ORDER COMPONENT 
export const UserContext = createContext(null);

