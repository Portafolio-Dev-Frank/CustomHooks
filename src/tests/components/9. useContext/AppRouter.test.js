import { shallow } from "enzyme"
import { AppRouter } from "../../../components/9. useContext/AppRouter"
import { UserContext } from "../../../components/9. useContext/UserContext"



describe('pruebas en <AppRouter></AppRouter>', () => {

    const user = {
        name: 'francisco',
        email: 'francisco.email.co'
    }

    const wrapper = shallow(
        <UserContext.Provider value={{user}}>
            <AppRouter></AppRouter>
        </UserContext.Provider>)

    test('debe de renderizar el <AppRouter></AppRouter> ', () => {
        expect(wrapper).toMatchSnapshot();
    })

})
