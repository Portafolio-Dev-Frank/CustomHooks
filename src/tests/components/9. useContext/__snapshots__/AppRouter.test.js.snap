// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <AppRouter></AppRouter> debe de renderizar el <AppRouter></AppRouter>  1`] = `<AppRouter />`;
